Makefile

```diff
@@ -34,9 +38,18 @@ CFLAGS += -fsigned-char $(DEVLIBS) \
 	-I/usr/include/SDL \
 	-I$(SDKSTAGE)/opt/vc/include -I$(SDKSTAGE)/opt/vc/include/interface/vcos/pthreads \
 	-I$(SDKSTAGE)/opt/vc/include/interface/vmcs_host/linux \
-	-I/usr/include/glib-2.0 -I/usr/lib/arm-linux-gnueabihf/glib-2.0/include \
-	-O3 -ffast-math -fno-builtin -fsingle-precision-constant \
-	-Wall -Wno-sign-compare -Wunused -Wpointer-arith -Wcast-align -Waggregate-return -Wshadow 
+	-I/usr/include/glib-2.0 -I/usr/lib/arm-linux-gnueabihf/glib-2.0/include
+
+# Platform specific
+ifeq ($(PLATFORM),rpi2)
+CFLAGS += -mcpu=cortex-a7 -mfpu=neon-vfpv4 -mfloat-abi=hard
+else
+CFLAGS += -march=armv6j -mfpu=vfp -mfloat-abi=hard
+endif
+
+CFLAGS += -O3 -ffast-math -fno-builtin -fsingle-precision-constant \
+	-fno-common -mstructure-size-boundary=32 -fweb -frename-registers \
+	-Wall -Wno-sign-compare -Wunused -Wpointer-arith -Wcast-align -Waggregate-return -Wshadow
 
```

Comment:
-mstructure-size-boundary=32 is set by default and  -fweb and -frename-registers are already enabled at our optimisation level.  I would prefer to see proof that fno-common is an improvement rather than guessing. the previous optimisation options were all removed due to the lack of understanding by whoever added them - most were already set at -O3